### PR TITLE
Type annotations for torchvision.ops

### PR DESCRIPTION
**Pull request by iamvukasin**
Addresses #2025

### CODE REVIEW

**Comment by fmassa** [comment]
Thanks for the PR!

Unfortunately, torchscript currently doesn't support `Union`, so for the type annotations we need to use only one of the overload that are supported (and in many cases we only support one for torchscript)

Also, if a function argument is not annotated, torchscript assumes that the argument is a Tensor, so for the functions that are missing type annotations you can safely annotate them with Tensor.

Can you look into fixing this?

File: torchvision/ops/boxes.py [hunk 1]
```diff
@@ -5,7 +5,7 @@
 
 
 @torch.jit.script
-def nms(boxes, scores, iou_threshold):
+def nms(boxes: Tensor, scores: Tensor, iou_threshold: float) -> Tensor:
     # type: (Tensor, Tensor, float) -> Tensor
```
Comment (fmassa) [inline torchvision/ops/boxes.py:9]: Can you remove the comment-style type annotations if you add the type annotations inline?

Comment (iamvukasin) [reply torchvision/ops/boxes.py:9]: Sure, I have it in the local repo, will push it soon.

**Comment by iamvukasin** [comment]
Thanks for the fast review! I have locally updated the code, but still have minor issues before pushing it.  

> Unfortunately, torchscript currently doesn't support Union, so for the type annotations we need to use only one of the overload that are supported (and in many cases we only support one for torchscript)

What to do with `check_roi_boxes_shape(boxes)` in `_utils.py` as `boxes` can be `List[Tensor]`, `Tuple[Tensor]` or `Tensor`? Which one is preferred?

> Also, if a function argument is not annotated, torchscript assumes that the argument is a Tensor, so for the functions that are missing type annotations you can safely annotate them with Tensor.

Is this for functions nested in `_register_custom_op()` in `_register_onnx_ops.py`, since they are only without annotations?

**Comment by fmassa** [comment]
> Is this for functions nested in _register_custom_op() in _register_onnx_ops.py, since they are only without annotations?

No, this is for all functions in torchvision which support torchscript.

> What to do with check_roi_boxes_shape(boxes) in _utils.py as boxes can be List[Tensor], Tuple[Tensor] or Tensor? Which one is preferred?

As I mentioned in my previous comment, if there is no type annotation for an argument, torchscript assumes the argument is a tensor, so for `check_roi_boxes_shape`, we should use `Tensor`

**Comment by fmassa** [comment]
Hi,

Test failures look weird, and might be due to the fact that CircleCI is running on your account. Let me have a look to understand what is going on.

cc @seemethere for thoughts

**Comment by seemethere** [comment]
@iamvukasin I think the reason this is happening is because you have enabled builds for your fork on CircleCI.

Could you go and disable builds on your fork 

1. Click this link -> https://app.circleci.com/settings/project/github/iamvukasin/vision
2. Click this button

Screen Shot 2020-06-30 at 9 25 25 AM

**Comment by iamvukasin** [comment]
@seemethere Definitely the problem was that the builds had been running with my CirceCI plan. I have stopped any further builds by following your guide.

**Comment by fmassa** [comment]
@iamvukasin Can you try rebasing this PR on top of master to try running new CI jobs?

File: torchvision/ops/misc.py [hunk 2]
```diff
@@ -46,7 +48,12 @@ class FrozenBatchNorm2d(torch.nn.Module):
     are fixed
     """
 
-    def __init__(self, num_features, eps=0., n=None):
+    def __init__(
+        self,
+        num_features: Tuple[int, ...],
```
Comment (fmassa) [inline torchvision/ops/misc.py:53]: nit: I would have expected `num_features` to be an `int`, not a tuple with a varying number of elements.

**Pull request merged**